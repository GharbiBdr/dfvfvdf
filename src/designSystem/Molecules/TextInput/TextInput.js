import React from 'react'
import PropTypes from 'prop-types'
import { View } from 'react-native'
import DSTextInput from '../../Atoms/DSTextInput/DSTextInput'
import DSLabel from '../../Atoms/DSLabel/DSLabel'
import Colors from '../../../assets/styles/Colors'
import Style from '../../../assets/styles/Style'
import DSIcon from '../../Atoms/DSIcon/DSIcon'

const TextInput = (props) => {
  const [focused, setFocused] = React.useState(false)
  const [passwordHidden, setPasswordHidden] = React.useState(true)

  const displayStyle1 = () => {
    return (
      <View style={Style.TextInputViewStyle1}>
        <DSTextInput
          {...(props.value !== undefined && { value: props.value })}
          value={props.value}

          onFocus={() => {
            setFocused(true)
            if (props.onFocus !== undefined) {
              props.onFocus()
            }
          }}
          onBlur={() => {
            setFocused(false)
            if (props.onBlur) {
              props.onBlur()
            }
          }}
          placeholder={focused ? '' : props.placeholder}
          autoCapitalize={props.autoCapitalize}
          autoCorrect={props.autoCorrect}
          editable={props.editable}
          keyboardType={props.keyboardType}
          {...(props.maxLength !== undefined && { maxLength: props.maxLength })}
          multiline={props.multiline}
          numberOfLines={props.numberOfLines}
          {...(props.onChange !== undefined && { onChange: props.onChange })}
          {...(props.onChangeText !== undefined && { onChangeText: props.onChangeText })}
          {...(props.onContentSizeChange !== undefined && { onContentSizeChange: props.onContentSizeChange })}
          {...(props.onEndEditing !== undefined && { onEndEditing: props.onEndEditing })}
          {...(props.onKeyPress !== undefined && { onKeyPress: props.onKeyPress })}
          {...(props.onLayout !== undefined && { onLayout: props.onLayout })}
          {...(props.onScroll !== undefined && { onScroll: props.onScroll })}
          {...(props.onSelectionChange !== undefined && { onSelectionChange: props.onSelectionChange })}
          {...(props.onSubmitEditing !== undefined && { onSubmitEditing: props.onSubmitEditing })}
          placeholderTextColor={props.placeholderTextColor}
          returnKeyType={props.returnKeyType}
          secureTextEntry={(props.type === 'password' && passwordHidden)}
          {...(props.selectionColor !== undefined && { selectionColor: props.selectionColor })}
          selectTextOnFocus={props.selectTextOnFocus}
          style={[Style.TextInputStyle1, props.multiline ? Style.textInputMultiline : null]}

        />
        {displayRightIcon()}
        {displayPasswordClosedShape()}
        {displayLabelStyle1()}
        {displayValididyCriterias()}
        {displayErrorMessageStyle1()}

      </View>
    )
  }

  const displayStyle2 = () => {
    return (
      <View style={Style.TextInputViewStyle2}>
        <DSTextInput
          {...(props.value !== undefined && { value: props.value })}
          value={props.value}
          onFocus={() => {
            setFocused(true)
            if (props.onFocus !== undefined) {
              props.onFocus()
            }
          }}
          onBlur={() => {
            setFocused(false)
          }}
          placeholder={focused ? '' : props.placeholder}
          autoCapitalize={props.autoCapitalize}
          autoCorrect={props.autoCorrect}
          editable={props.editable}
          keyboardType={props.keyboardType}
          {...(props.maxLength !== undefined && { maxLength: props.maxLength })}
          multiline={props.multiline}
          numberOfLines={props.numberOfLines}
          {...(props.onChange !== undefined && { onChange: props.onChange })}
          {...(props.onChangeText !== undefined && { onChangeText: props.onChangeText })}
          {...(props.onContentSizeChange !== undefined && { onContentSizeChange: props.onContentSizeChange })}
          {...(props.onEndEditing !== undefined && { onEndEditing: props.onEndEditing })}
          {...(props.onKeyPress !== undefined && { onKeyPress: props.onKeyPress })}
          {...(props.onLayout !== undefined && { onLayout: props.onLayout })}
          {...(props.onScroll !== undefined && { onScroll: props.onScroll })}
          {...(props.onSelectionChange !== undefined && { onSelectionChange: props.onSelectionChange })}
          {...(props.onSubmitEditing !== undefined && { onSubmitEditing: props.onSubmitEditing })}
          placeholderTextColor={props.placeholderTextColor}
          returnKeyType={props.returnKeyType}
          secureTextEntry={(props.type === 'password' && passwordHidden)}
          {...(props.selectionColor !== undefined && { selectionColor: props.selectionColor })}
          selectTextOnFocus={props.selectTextOnFocus}
          style={Style.TextInputStyle2}

        />
        {displayRightIcon()}
      </View>
    )
  }

  const displayLabelStyle1 = () => {
    if (props.label && (focused || props.value)) {
      return (
        <DSLabel
          {...(props.onLabelLayout !== undefined && { onLayout: props.onLabelLayout })}
          {...(props.onLabelLongPress !== undefined && { onLongPress: props.onLabelLongPress })}
          {...(props.onLabelPress !== undefined && { onPress: props.onLabelPress })}
          {...(props.onLabelTextLayout !== undefined && { onTextLayout: props.onLabelTextLayout })}
          selectable={props.selectable}
          style={Style.TextInputLabelStyle1}
        >
          {props.label}
        </DSLabel>
      )
    }
  }

  const displayRightIcon = () => {
    const color = (focused || props.value) ? Colors.primary : Colors.grey1
    if (props.rightIcon || props.type === 'password') {
      return (
        <DSIcon
          onPress={props.onRightIconPress ? props.onRightIconPress : props.type === 'password' ? togglePasswordMode : null}
          {...(props.onRightIconPressIn !== undefined && { onPressIn: props.onRightIconPressIn })}
          {...(props.onRightIconPressOut !== undefined && { onPressOut: props.onRightIconPressOut })}
          {...(props.onRightIconLongPress !== undefined && { onLongPress: props.onRightIconLongPress })}
          style={Style.TextInputRightIcon}
          name={props.rightIcon ? props.rightIcon : 'eye'}
          color={color} size="small"
        />
      )
    }
    return null
  }
  const displayPasswordClosedShape = () => {
    return passwordHidden
      ? null
      : (
        <View style={(focused || props.value) ? Style.passwordEyeClosedFocused : Style.passwordEyeClosed} />
      )
  }

  const togglePasswordMode = () => {
    if (props.type === 'password') {
      setPasswordHidden(!passwordHidden)
    }
  }

  const displayErrorMessageStyle1 = () => {
    if (props.isInvalid) {
      return (
        <DSLabel style={Style.textInputErrorMessageStyle1}>{props.errorMessage}</DSLabel>
      )
    }
  }

  const displayValididyCriterias = () => {
    if (props.valididtyCriterias.length > 0) {
      const display = []
      for (let i = 0; i < props.valididtyCriterias.length; i++) {
        display.push(<DSLabel style={props.valididtyCriterias[i].valid ? Style.TextInputValidityCriteriaValid : Style.TextInputValidityCriteria}>
          {props.valididtyCriterias[i].text}
        </DSLabel>)
      }
      return (
        <View style={Style.textInputValidityCriteriasContainerStyle1}>
          {display}
        </View>
      )
    }
  }

  if (props.styleType === 'style-1') {
    return displayStyle1()
  } else if (props.styleType === 'style-2') {
    return displayStyle2()
  }
  return null
}

TextInput.defaultProps = {
  styleType: 'style-1',
  label: '',
  value: '',
  placeholder: '',
  autoCapitalize: 'sentences', // 'none', 'sentences', 'words', 'characters'
  autoCorrect: false,
  editable: true,
  keyboardType: 'default', // 'default', 'number-pad', 'decimal-pad', 'numeric', 'email-address', 'phone-pad'
  multiline: false,
  numberOfLines: 5,
  placeholderTextColor: Colors.grey1,
  returnKeyType: 'done', // 'done', 'go', 'next', 'search', 'send',
  selectTextOnFocus: false,
  type: 'text', // 'text', 'password',
  isInvalid: false,
  errorMessage: '',
  valididtyCriterias: []
}

TextInput.propTypes = {
  styleType: PropTypes.string,
  value: PropTypes.string,
  label: PropTypes.string,
  defaultValue: PropTypes.string,
  placeholder: PropTypes.string,
  autoCapitalize: PropTypes.string,
  autoCorrect: PropTypes.bool,
  editable: PropTypes.bool,
  keyboardType: PropTypes.string,
  maxLength: PropTypes.number,
  multiline: PropTypes.bool,
  numberOfLines: PropTypes.number,
  onChange: PropTypes.func,
  onChangeText: PropTypes.func,
  onContentSizeChange: PropTypes.func,
  onEndEditing: PropTypes.func,
  onKeyPress: PropTypes.func,
  onLayout: PropTypes.func,
  onScroll: PropTypes.func,
  onSelectionChange: PropTypes.func,
  onSubmitEditing: PropTypes.func,
  placeholderTextColor: PropTypes.string,
  returnKeyType: PropTypes.string,
  selectionColor: PropTypes.string,
  selectTextOnFocus: PropTypes.bool,
  onLabelLayout: PropTypes.func,
  onLabelLongPress: PropTypes.func,
  onLabelPress: PropTypes.func,
  onLabelTextLayout: PropTypes.func,
  rightIcon: PropTypes.string,
  type: PropTypes.oneOf(['text', 'password']),
  onRightIconPress: PropTypes.func,
  onRightIconLongPress: PropTypes.func,
  onRightIconPressIn: PropTypes.func,
  onRightIconPressOut: PropTypes.func,
  isInvalid: PropTypes.bool,
  errorMessage: PropTypes.string,
  valididtyCriterias: PropTypes.array,
  onFocus: PropTypes.func
}

export default TextInput
