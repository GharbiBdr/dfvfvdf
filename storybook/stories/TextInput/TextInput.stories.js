import { action } from '@storybook/addon-actions';
import { text,boolean,select,number,color,array } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import React from 'react';
import CenterView from '../CenterView';
import {TextInput,Container} from "../../../src/designSystem/"

const Test1=(props)=>{
    const [value,setValue]=React.useState(props.value)

    return (
        <Container>
            <CenterView>
            <TextInput
                styleType={select('styleType',['style-1','style-2'],'style-1','Options')}
                label={text('label','label','Options')}
                isInvalid={boolean('isInvalid',false,'Options')}
                value={value}
                errorMessage="this is an error message"
                onChangeText={(val)=>{
                    action('onChangeText')
                    setValue(val)
                }}
                onChange={action('onChange')}
                onContentSizeChange={action('onContentSizeChange')}
                onEndEditing={action('onEndEditing')}
                onKeyPress={action('onKeyPress')}
                onLayout={action('onLayout')}
                onScroll={action('onScroll')}
                onLabelLayout={action('onLabelLayout')}
                onLabelLongPress={action('onLabelLongPress')}
                onLabelPress={action('onLabelPress')}
                onSelectionChange={action('onSelectionChange')}
                onSubmitEditing={action('onSubmitEditing')}
                onLabelTextLayout={action('onLabelTextLayout')}
                onRightIconPress={action('onRightIconPress')}
                onRightIconLongPress={action('onRightIconLongPress')}
                onRightIconPressIn={action('onRightIconPressIn')}
                onRightIconPressOut={action('onRightIconPressOut')}
                placeholder={text('placeholder','label','Options')}
                autoCapitalize={select('autoCapitalize',['none','sentences','words','characters'],'sentences','Options')}
                autoCorrect={boolean("autoCorrect",false,'Options')}
                editable={boolean("editable",true,'Options')}
                keyboardType={select("keyboardType",['default','number-pad','decimal-pad','numeric','email-address','phone-pad'],'default','Options')}
                multiline={boolean("multiline",false,'Options')}
                numberOfLines={number('numberOfLines',5,{},'Options')}
                placeholderTextColor={color('placeholderTextColor','#ADB4BB','Options')}
                returnKeyType={select('returnKeyType',['done', 'go', 'next', 'search', 'send'],'done','Options')}
                selectTextOnFocus={boolean("selectTextOnFocus",false,'Options')}
                type={select('type',['text','password'],'text','Options')}
                valididtyCriterias={array('valididtyCriterias',[{text:"> 8 caractères" ,valid:false},{text:"1 majuscule" ,valid:true},{text:"> 1 caractère spécial" ,valid:false}],',','Options')}
                selectionColor={color('selectionColor','#ADB4BB','Options')}
                selectTextOnFocus={boolean("selectTextOnFocus",false,'Options')}
                rightIcon={
                    select(
                        'rightIcon',
                        [
                            '',
                            'arrow-down',
                            'arrow-left',
                            'arrow-right',
                            'arrow-up',
                            'avatar',
                            'camera',
                            'check',
                            'contact',
                            'copy',
                            'download',
                            'eye',
                            'go-back',
                            'home',
                            'list',
                            'lot',
                            'menu',
                            'person',
                            'plus',
                            'Rectangle',
                            'search',
                            'settings',
                            'x',
                        ],
                        '',
                        'Options'
                    )
                }
            />
            </CenterView>
        </Container>
    )
}

storiesOf('TextInput', module)
.addDecorator((getStory) => <CenterView>{getStory()}</CenterView>)
.add('style 1', () => (
    <Test1
        onChangeText={text('onChangeText', 'a callback function','Callbacks')}
        onChange={text('onChange', 'a callback function','Callbacks')}
        onContentSizeChange={text('onContentSizeChange', 'a callback function','Callbacks')}
        onEndEditing={text('onEndEditing', 'a callback function','Callbacks')}
        onKeyPress={text('onKeyPress', 'a callback function','Callbacks')}
        onLayout={text('onLayout', 'a callback function','Callbacks')}
        onScroll={text('onScroll', 'a callback function','Callbacks')}
        onSelectionChange={text('onSelectionChange', 'a callback function','Callbacks')}
        onSubmitEditing={text('onSubmitEditing', 'a callback function','Callbacks')}
        onLabelLayout={text('onLabelLayout', 'a callback function','Callbacks')}
        onLabelLongPress={text('onLabelLongPress', 'a callback function','Callbacks')}
        onLabelPress={text('onLabelPress', 'a callback function','Callbacks')}
        onLabelTextLayout={text('onLabelTextLayout', 'a callback function','Callbacks')}
        onRightIconPress={text('onRightIconPress', 'a callback function','Callbacks')}
        onRightIconLongPress={text('onRightIconLongPress', 'a callback function','Callbacks')}
        onRightIconPressIn={text('onRightIconPressIn', 'a callback function','Callbacks')}
        onRightIconPressOut={text('onRightIconPressOut', 'a callback function','Callbacks')}
        value={text('value','test','Options')}
    />
  ))