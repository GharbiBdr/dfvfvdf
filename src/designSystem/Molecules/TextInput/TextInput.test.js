import React from 'react';
import renderer from 'react-test-renderer'
import TextInput from "./TextInput"
import DSTextInput from "../../Atoms/DSTextInput/DSTextInput"
import DSIcon from "../../Atoms/DSIcon/DSIcon"
import { shallow } from 'enzyme'

describe("TextInput tests",()=>{
    it('test render text input snapshot',()=>{
        const wrapper=renderer.create(<TextInput value="test" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })

    it('test text input value',()=>{
        const wrapper=shallow(<TextInput value="test" lable="Label" placeholder="Placeholder" />)
        expect(wrapper.find(DSTextInput).first().props().value===`test`)
        .toEqual(true)
    })

    it('test render password text input snapshot',()=>{
        const wrapper=renderer.create(<TextInput type='password' value="" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })

    it('test password type',()=>{
        const wrapper=shallow(<TextInput type='password' value="" lable="Label" placeholder="Placeholder" />)
        expect(wrapper.find(DSTextInput).first().props().secureTextEntry===true)
        .toEqual(true)
    })

    it('test render text input width right icon camera snapshot',()=>{
        const wrapper=renderer.create(<TextInput rightIcon="camera" value="" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })

    it('test icon shallow',()=>{
        const wrapper=shallow(<TextInput rightIcon="camera" />)
        expect(wrapper.find(DSIcon).first().props().name==="camera")
        .toEqual(true)
    })

    it('test icon click',()=>{
        const press1=jest.fn()
        const wrapper=renderer.create(<TextInput rightIcon="camera" onRightIconPress={press1} />)
        wrapper.root.findAllByType(DSIcon)[0].props.onPress()
        expect(press1.mock.calls.length).toEqual(1)
    })

    it('test render multiliane text input snapshot',()=>{
        const wrapper=renderer.create(<TextInput multiline value="" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
    it('test render text input with error message snapshot',()=>{
        const wrapper=renderer.create(<TextInput errorMessage="error" value="" lable="Label" placeholder="Placeholder" />)
        .toJSON()
        expect(wrapper).toMatchSnapshot()
    })
    
})