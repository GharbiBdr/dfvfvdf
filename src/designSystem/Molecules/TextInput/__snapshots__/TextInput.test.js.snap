// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput tests test render multiliane text input snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 22,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    keyboardType="default"
    multiline={true}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    placeholderTextColor="#ADB4BB"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={false}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FDFDFD",
          "borderColor": "#E5E5E5",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "#FF8577",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 48,
          "letterSpacing": 1,
          "lineHeight": 14,
          "padding": 18,
          "width": "100%",
        },
        Object {
          "height": 150,
        },
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;

exports[`TextInput tests test render password text input snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 22,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    placeholderTextColor="#ADB4BB"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={true}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FDFDFD",
          "borderColor": "#E5E5E5",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "#FF8577",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 48,
          "letterSpacing": 1,
          "lineHeight": 14,
          "padding": 18,
          "width": "100%",
        },
        null,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": 14,
        "top": 14,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ADB4BB",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`TextInput tests test render text input snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 22,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    placeholderTextColor="#ADB4BB"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={false}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FDFDFD",
          "borderColor": "#E5E5E5",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "#FF8577",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 48,
          "letterSpacing": 1,
          "lineHeight": 14,
          "padding": 18,
          "width": "100%",
        },
        null,
      ]
    }
    underlineColorAndroid="transparent"
    value="test"
  />
</View>
`;

exports[`TextInput tests test render text input width right icon camera snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 22,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    placeholderTextColor="#ADB4BB"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={false}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FDFDFD",
          "borderColor": "#E5E5E5",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "#FF8577",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 48,
          "letterSpacing": 1,
          "lineHeight": 14,
          "padding": 18,
          "width": "100%",
        },
        null,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "position": "absolute",
        "right": 14,
        "top": 14,
      }
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#ADB4BB",
            "fontSize": 20,
          },
          undefined,
          Object {
            "fontFamily": "icomoon",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          Object {},
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`TextInput tests test render text input with error message snapshot 1`] = `
<View
  style={
    Object {
      "marginTop": 22,
      "width": "100%",
    }
  }
>
  <TextInput
    allowFontScaling={false}
    autoCapitalize="sentences"
    autoCorrect={false}
    defaultValue=""
    editable={true}
    keyboardType="default"
    multiline={false}
    numberOfLines={5}
    onBlur={[Function]}
    onFocus={[Function]}
    placeholder="Placeholder"
    placeholderTextColor="#ADB4BB"
    rejectResponderTermination={true}
    returnKeyType="done"
    secureTextEntry={false}
    selectTextOnFocus={false}
    style={
      Array [
        Object {
          "backgroundColor": "#FDFDFD",
          "borderColor": "#E5E5E5",
          "borderRadius": 10,
          "borderWidth": 1,
          "color": "#FF8577",
          "fontFamily": "Roboto-Bold",
          "fontSize": 12,
          "fontWeight": "700",
          "height": 48,
          "letterSpacing": 1,
          "lineHeight": 14,
          "padding": 18,
          "width": "100%",
        },
        null,
      ]
    }
    underlineColorAndroid="transparent"
    value=""
  />
</View>
`;
